pkg/cmd/pr/status/status.go

```diff
@@ -211,7 +211,8 @@ func prSelectorForCurrentBranch(gitClient *git.Client, baseRepo ghrepo.Interface
 	if branchOwner != "" {
 		if branchConfig.Push != "" {
 			selector = strings.TrimPrefix(branchConfig.Push, branchConfig.RemoteName+"/")
-		} else if strings.HasPrefix(branchConfig.MergeRef, "refs/heads/") {
+		} else if pushDefault, _ := gitClient.Config(context.Background(), "push.default"); (pushDefault == "upstream" || pushDefault == "tracking") &&
+			strings.HasPrefix(branchConfig.MergeRef, "refs/heads/") {
 			selector = strings.TrimPrefix(branchConfig.MergeRef, "refs/heads/")
 		}
```

Comment:
Don't worry, it will only get worse /s

Jokes aside, IIRC my original plan was to see if I could do a little refactoring when I got to fixing `gh pr create`. Now my short memory has got the best of me and I can't remember what issues `gh pr create` has exactly hehe. But if anyone's got some pointers in mind, feel free to share, and I can check them out when/if I find some time to get back on this :)

Reply:
Sure: something to follow-up on later. Maybe I'll take a stab sometime